docs/Bridging_Assets.md

```diff
@@ -0,0 +1,158 @@
+     * @param account destination address
+     * @param amount amount of tokens withdrawn
+     */
+    function withdraw(address account, uint256 amount) external override {
+```
+
+This initiates the withdrawal process (i.e., tokens can later be claimed via the Outbox.)
+
+#### Custom Token Bridging
+
+The Canonical Token Bridge also allows for pairing an ERC20 on L1 with a custom token contract on Arbitrum.
+
+**It is important to note**: The Canonical Token Bridge can't account for every possible token contract; if your custom token logic is [sufficiently weird](https://quoteinvestigator.com/2018/12/25/universe/#:~:text=Professor%20J.%20B.%20S.%20Haldane%20once%20shrewdly,the%20ultimate%20queerness%20of%20time.), it may require it's own, custom, taylor-made bridge as well. The implications of a particular custom token should be thought through carefully; feel free to [reach out to us](https://discord.gg/ZpZuw7p). With that said, the steps for creating a custom token pairing in the canonical bridge are as follows.
+
+#### Setting up a custom token pairing
+
+**1. Deploy L2 custom token contract**
+
+Start by deploying the L2 token contract [directly onto Arbitrum](Contract_Deployment.md). A custom L2 contract should conform to the minimal IArbToken interface:
+
+```sol
+/**
+ * @title Minimum expected interface for L2 token that interacts with the L2 token bridge (this is the interface necessary
+ * for a custom token that interacts with the bridge, see TestArbCustomToken.sol for an example implementation).
+ */
+interface IArbToken {
+    /**
+     * @notice should increase token supply by amount, and should (probably) only be callable by the L1 bridge.
+     */
+    function bridgeMint(address account, uint256 amount) external;
+
+    /**
+     * @notice should decrease token supply by amount, and should (probably) only be callable by the L1 bridge.
+     */
+    function bridgeBurn(address account, uint256 amount) external;
+
+    /**
+     * @notice withdraw user tokens from L2 to the L1
+     */
+    function withdraw(address account, uint256 amount) external;
+}
+```
+
+**2. Deploy L1 custom token contract**
+
+The L1 contract should conform to the minimal `ICustomToken` interface:
+
+```sol
+/**
+ * @title Minimum expected interface for L1 custom token (see TestCustomTokenL1.sol for an example implementation)
+ */
+interface ICustomToken {
+    /**
+     * @notice Should make an external call to EthERC20Bridge.registerCustomL2Token
+     */
+    function registerTokenOnL2(
+        address l2CustomTokenAddress,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid,
+        address refundAddress
+    ) external virtual;
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external virtual returns (bool);
+
+    function balanceOf(address account) external view virtual returns (uint256);
+}
+```
+
+**3. Register L1/L2 custom pairing**
```

Comment:
Should we mention the TMT logic? The L2 bridge doesn't check if there is a custom token actually deployed there until it tries interacting. 

The implications of this are: the address oracle from L1/L2 are always the same, but if there isn't a custom token in L2 you get the TMT

Reply:
I thought about mentioning this but I almost want to avoid bringing it up to just encourage SOP. I'll try a middle ground